curves/bn254/src/lib.rs

```diff
@@ -247,31 +319,18 @@ mod target_arch {
         let ele_len = input.len().saturating_div(ALT_BN128_PAIRING_ELEMENT_LEN);
 
         let mut vec_pairs: Vec<(G1, G2)> = Vec::new();
-        for i in 0..ele_len {
+        let mut current_byte_index = 0_usize;
+        for _ in 0..ele_len {
```

Comment:
Instead of using `(p|q)_bytes_start`/`(p|q)_bytes_end` counters, we could explicitly iterate over chunks:

```rust
    for chunk in input.chunks(ALT_BN128_PAIRING_ELEMENT_LEN) {
        let (p_bytes, q_bytes) = chunk.split_at(G1_POINT_SIZE);
        
        let g1 = PodG1::from_be_bytes(p_bytes)?.try_into()?;
        let g2 = PodG2::from_be_bytes(q_bytes)?.try_into()?;
        
        vec_pairs.push((g1, g2));
    }
```

I've been even thinking whether using `map` and `collect` on the `chunks` would make sense, but handling errors could be a bit challenging.

Reply:
Yeah the snippet that you wrote seems to be the cleanest implementation. Thanks!